From: Wang Ming <machel@vivo.com>
Subject: lib: error-inject: remove error checking for debugfs_create_dir()
Date: Wed, 19 Jul 2023 14:44:10 +0000

It is expected that most callers should _ignore_ the errors return by
debugfs_create_dir() in ei_debugfs_init().

Link: https://lkml.kernel.org/r/20230719144355.6720-1-machel@vivo.com
Signed-off-by: Wang Ming <machel@vivo.com>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/error-inject.c |    2 --
 1 file changed, 2 deletions(-)

--- a/lib/error-inject.c~lib-error-inject-remove-error-checking-for-debugfs_create_dir
+++ a/lib/error-inject.c
@@ -217,8 +217,6 @@ static int __init ei_debugfs_init(void)
 	struct dentry *dir, *file;
 
 	dir = debugfs_create_dir("error_injection", NULL);
-	if (!dir)
-		return -ENOMEM;
 
 	file = debugfs_create_file("list", 0444, dir, NULL, &ei_fops);
 	if (!file) {
_
